Source/Shaders/Builtin/Functions/approximateSphericalCoordinates.glsl

```diff
@@ -0,0 +1,58 @@
+// Based on Michal Drobot's approximation from ShaderFastLibs, which in turn is based on
+// "Efficient approximations for the arctangent function," Rajan, S. Sichun Wang Inkol, R. Joyal, A., May 2006.
+// Adapted from ShaderFastLibs under MIT License.
+//
+// Chosen for the following characteristics over range [0, 1]:
+// - basically no error at 0 and 1, important for getting around range limit (naive atan2 via atan requires infinite range atan)
+// - no visible artifacts from first-derivative discontinuities, unlike latitude via range-reduced sqrt asin approximations (at equator)
+//
+// The original code is x * (-0.1784 * abs(x) - 0.0663 * x * x + 1.0301);
+// Removed the abs() in here because it isn't needed, the input range is guaranteed as [0, 1] by how we're approximating atan2.
+float fastApproximateAtan01(float x) {
+    return x * (-0.1784 * x - 0.0663 * x * x + 1.0301);
+}
+
+// Used in place of ternaries to trade some math for branching
+float branchFree(bool comparison, float a, float b) {
```

Comment:
If this is generally useful to other shaders, perhaps put in its own file and rename to `czm_branchFreeTernary`.